src/main/java/com/cloudant/client/api/DesignDocumentManager.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2015 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.client.api;
+
+import static com.cloudant.client.org.lightcouch.internal.CouchDbUtil.assertNotEmpty;
+import static com.cloudant.client.org.lightcouch.internal.URIBuilder.buildUri;
+
+import com.cloudant.client.api.model.DesignDocument;
+import com.cloudant.client.org.lightcouch.CouchDatabase;
+import com.cloudant.client.org.lightcouch.CouchDatabaseBase;
+import com.cloudant.client.org.lightcouch.NoDocumentException;
+import com.cloudant.client.org.lightcouch.Response;
+import com.cloudant.client.org.lightcouch.internal.CouchDbUtil;
+import com.google.gson.Gson;
+
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Provides API to work with design documents.
+ * <h3>Usage Example:</h3>
+ * <pre>
+ * // read from javascript design document file
+ * {@code
+ * DesignDocument design = DesignDocumentManager.fromFile("../example.js");
+ * }
+ *
+ * // read from directory of design documents
+ * {@code
+ * List<DesignDocument>DesignDocument design =
+ *                  DesignDocumentManager.fromDirectory("../design-files/");
+ * }
+ *
+ * // sync with the database
+ * {@code
+ * db.getDesignDocumentManager().put(designDoc);
+ * }
+ *
+ * // sync multiple design documents with the database
+ * {@code
+ * db.getDesignDocumentManager().put(listDesignDocuments);
+ * }
+ *
+ * // read from the database
+ * {@code
+ * DesignDocument designDoc2 = db.getDesignDocumentManager().get("_design/example");
+ * }
+ * </pre>
+ *
+ * @see {@link Database#getDesignDocumentManager()} to access the API.
+ * @see DesignDocument
+ * @since 2.0.0
+ */
+public class DesignDocumentManager {
+
+    private URI dbUri;
+    private CouchDatabaseBase dbc;
+    private CloudantClient client;
+
+    DesignDocumentManager(CouchDatabase db, CloudantClient client) {

```

Comment:
Can this just take a `Database` object instead and then all the methods can be much simpler, calling `db.save()`,  `db.find()`, `db.remove()` rather than having to build all these URIs again?